rust/src/main.rs

```diff
@@ -8,9 +8,14 @@ mod cli;
 async fn main() -> Result<()> {
     let args: Args = cli::parse_args().await;
 
-    let client = Vault::new(None, args.region.as_deref())
-        .await
-        .with_context(|| "Failed to create vault.".to_string())?;
+    let client = Vault::new(
+        args.vault_stack.as_deref(),
+        args.region.as_deref(),
+        args.bucket.as_deref(),
+        args.key_arn.as_deref(),
+    )
```

Comment:
I would prefer to implement `from_bucket_and_key` than would as mentioned previously, that would create the `Vault` with that method if both of those values exist, otherwise have the signature of `new` stay the same but parse the `env` variables inside the library.
Refactoring/improving the public interface with breaking changes should be done later in #216 .